### PR TITLE
Q378 Add KthSmallestElementInASortedMatrix.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: KthSmallestElementInASortedMatrix.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                self.heapAdd(heap, (matrix[x][y + 1], x, y + 1))
+            k -= 1
+        return self.heapPop(heap)[0]
+    
+    def heapAdd(self, heap, item):
+        heap.append(item)
+        self.bubbleUp(heap, len(heap) - 1)
+    
+    def heapPop(self, heap):
+        length = len(heap) - 1
```
Comment (nkukarl) [inline KthSmallestElementInASortedMatrix.py]: Why do you need this line?
Can just do
```
heap[0], heap[-1] = heap[-1], heap[0]
```
on the next line

File: KthSmallestElementInASortedMatrix.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        if not matrix:
+            return 0
+        m = len(matrix)
+        n = len(matrix[0])
+        heap = []
+        for i in range(m):
+            self.heapAdd(heap, (matrix[i][0], i, 0))
+        while k > 1:
+            minItem = self.heapPop(heap)
```
Comment (nkukarl) [inline KthSmallestElementInASortedMatrix.py]: You can do
`_, x, y = self.heapPop(heap)`

File: KthSmallestElementInASortedMatrix.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        :type matrix: List[List[int]]
+        :type k: int
+        :rtype: int
+        """
+        if not matrix:
+            return 0
+        m = len(matrix)
+        n = len(matrix[0])
+        heap = []
```
Comment (nkukarl) [inline KthSmallestElementInASortedMatrix.py]: Apparently, this is not OOP.
Should define a class Heap and define its methods.

File: KthSmallestElementInASortedMatrix.py (hunk 1 above)
Comment (peinanteng) [reply KthSmallestElementInASortedMatrix.py]: sure, will do this.

File: KthSmallestElementInASortedMatrix.py (hunk 3 above)
Comment (peinanteng) [reply KthSmallestElementInASortedMatrix.py]: Comments has been addressed.

File: KthSmallestElementInASortedMatrix.py (hunk 2 above)
Comment (peinanteng) [reply KthSmallestElementInASortedMatrix.py]: sure, will do

**Pull request merged**